classes/Task/Upgrade/BackupDb.php

```diff
@@ -271,39 +240,146 @@ public function run(): int
+
+        return ExitCode::SUCCESS;
+    }
+
+    /**
+     * @param array<array<string, string>> $listOfTables
+     *
+     * @internal Method is public for unit tests
+     *
+     * @return string[]
+     */
+    public function filterTablesToSync(array $listOfTables): array
+    {
+        return array_filter(array_map('current', $listOfTables), function ($table) {
+            // Skip tables which do not start with _DB_PREFIX_
+            if (strlen($table) <= strlen(_DB_PREFIX_) || strncmp($table, _DB_PREFIX_, strlen(_DB_PREFIX_)) !== 0) {
+                return false;
             }
-            $this->stepDone = true;
-            // reset dbStep at the end of this step
-            $this->container->getState()->setDbStep(0);
 
-            $this->logger->info($this->translator->trans('Database backup done in filename %s. Now upgrading files...', [$this->container->getState()->getBackupName()]));
-            $this->next = 'upgradeFiles';
+            // Ignore stat tables
+            if (in_array($table, $this->getTablesToIgnore())) {
+                return false;
+            }
 
-            return ExitCode::SUCCESS;
+            return true;
+        });
+    }
+
+    /**
+     * @return string[]
+     */
+    private function getTablesToIgnore(): array
+    {
+        return [
+            _DB_PREFIX_ . 'connections',
+            _DB_PREFIX_ . 'connections_page',
+            _DB_PREFIX_ . 'connections_source',
+            _DB_PREFIX_ . 'guest',
+            _DB_PREFIX_ . 'statssearch',
+        ];
+    }
+
+    // MANAGEMENT OF BACKUP FILE RESOURCE
+
+    /**
+     * @return resource
+     *
+     * @throws Exception if file already exists or cannot be written
+     */
+    private function openPartialBackupFile(string $backupfile)
+    {
+        // Figure out what compression is available and open the file
+        if (file_exists($backupfile)) {
+            throw (new UpgradeException($this->translator->trans('Backup file %s already exists. Operation aborted.', [$backupfile])))->setSeverity(UpgradeException::SEVERITY_ERROR);
+        }
+
+        if (function_exists('bzopen')) {
+            $backupfile .= '.bz2';
+            $fp = bzopen($backupfile, 'w');
+        } elseif (function_exists('gzopen')) {
+            $backupfile .= '.gz';
+            $fp = gzopen($backupfile, 'w');
+        } else {
+            $fp = fopen($backupfile, 'w');
+        }
+
+        if ($fp === false) {
+            throw (new UpgradeException($this->translator->trans('Unable to create backup database file %s.', [addslashes($backupfile)])))->setSeverity(UpgradeException::SEVERITY_ERROR);
+        }
+
+        return $fp;
+    }
+
+    private function isRemainingTimeEnough(int $elapsedTime): bool
+    {
+        $timeAllowed = (int) @ini_get('max_execution_time');
+
+        if ($timeAllowed <= 0) {
+            return true;
         }
+
+        return $elapsedTime < $timeAllowed - 5;
```

Comment:
I think it would me great to add a comment to explain the -5

Reply:
```suggestion
        // Remove 5 seconds on the allowed time to make sure we have time to save data and close files .
        return $elapsedTime < $timeAllowed - 5;
```